File: Eunleelee/1월 4일/연속_부분_수열_합의_개수.py
from collections import deque

def solution(elements):
    total_sum = list()  # 합으로 나올 수 있는 모든 수를 저장
    total_sum.extend(elements)
    stack = deque()

    for i in range(2, len(elements)):
        circle = deque(elements + elements[:i-1])  # 이렇게 하면 모든 합의 경우의 수를 구할 수 있음
        while circle:
            stack.append(circle.popleft())
            if len(stack) == i:
                total_sum.append(sum(stack))
                stack.popleft()
        stack.clear()  # 찌꺼기 수가 남아있을 수 있음을 고려

    total_sum.append(sum(elements))  # 마지막으로 모든 수를 합한 값을 추가
    answer = len(list(set(total_sum)))  # 중복되는 값을 제거 후, 배열의 길이를 구함
            
    return answer

print(solution([7, 9, 1, 1, 4]))

File: yht0827/1월/1월 4일/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
	public int solution(int[] elements) {
		Set<Integer> set = new HashSet<>();

		int k = 0;
		while (k <= elements.length) {
			for (int i = 0; i < elements.length; i++) {
				int sum = 0;

				for (int j = i; j < i + k; j++) {
					sum += elements[j % elements.length];
				}
				set.add(sum);
			}
			k++;
		}
		return set.size() - 1;
	}
}